libs/wingsdk/src/core/tree.ts

```diff
@@ -68,6 +69,37 @@ export interface DisplayInfo {
    * @default - no source information
    */
   readonly sourceModule?: string;
+
+  /**
+   * UI components to display for this resource.
+   * @default - no UI components
+   */
+  readonly uiComponents?: any[]; // UIComponent
+}
+
+/** @internal */
+export type UIComponent = UIField | UISection | UIButton;
```

Comment:
Maybe also add `UILabel` and `UILink`?

Reply:
What would be the API for UILabel and UILink?